lib/instance.js

```diff
@@ -640,16 +641,24 @@ Instance.prototype.save = function(options) {
         self.dataValues[createdAtAttr] = values[self.Model.rawAttributes[createdAtAttr].field || createdAtAttr] = self.Model.$getDefaultTimestamp(createdAtAttr) || now;
       }
 
+      var tableName;
+      if (self.$options.$schema !== self.Model.$schema ||
+        self.$options.$schemaDelimiter !== self.Model.$schemaDelimiter) {
+        tableName = self.Model.schema(self.$options.$schema, self.$options.$schemaDelimiter).getTableName(options);
+      } else {
```

Comment:
Not sure i understand, couldn't we just:

```js
var clone = Object.create(Model);
clone.instance = function() {
   Model.Instance.apply(this, arguments);
};
clone.Instance.prototype = Object.clone(Model.Instance.prototype);
clone.Instance.prototype.$Model = clone;
```

`Model.schema` should perhaps then have a cache, LRU perhaps.
I would greatly prefer it to having 3-4 places with special handling code.

/cc @janmeier